[ignoreme] experiment with `go test -json`

tbg opened the pull request:
See https://github.com/cockroachdb/go-test-teamcity/issues/5 for context.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/30406)
<!-- Reviewable:end -->

tbg committed:
failsuite: add package to create corpus for failed test output

This package serves to make it easy to improve test coverage for
go-test-teamcity in the future as problems arise, by opening a PR that
edits the CI scripts to include the COCKROACH_FAILSUITE environment
variable, triggering various interesting test failures and data races in
these newly introduced packages. The created artifacts can then be
incorporated into the test suite for go-test-teamcity (as was already
done in https://github.com/cockroachdb/go-test-teamcity/pull/9).

Release note: None

tbg committed:
[wip,dnm]: trigger the failsuite

Release note: None

jordanlewis commented:
@tbg this is dead right?
